packages/transaction-controller/src/TransactionController.ts

```diff
@@ -982,9 +982,19 @@ export class TransactionController extends BaseController<
 
         /* istanbul ignore next */
         if (txObj?.blockNumber) {
-          meta.status = TransactionStatus.confirmed;
-          this.hub.emit(`${meta.id}:confirmed`, meta);
-          return [meta, true];
+          // transactions can be added to a block and still fail, so we need to check the transaction status before emitting the confirmed event
+          const verifyTransactionStatus =
+            await this.checkTxReceiptStatusIsFailed(transactionHash);
+          if (verifyTransactionStatus) {
+            const error = new Error(
+              'Transaction failed. Error encountered during contract execution [out of gas]',
```

Comment:
Is this error message sufficient or valid? Here's a link to a failed [transaction](https://goerli.etherscan.io/tx/0xb5e68e55fc2a199c169e173fcf7644b5e6273d3d0a2492823f2093e4203b2670)

Reply:
I updated this error message to match the message in line [957](https://github.com/MetaMask/core/blob/7163988613a645bd94a8ee456c3660cafb305b79/packages/transaction-controller/src/TransactionController.ts#L957)